firestore.rules

```diff
@@ -16,7 +11,22 @@ service cloud.firestore {
       resource.data.authorId == request.auth.uid);
     }
 
-    // Add new rules here
+    //Allow authenticated user to view, create, update and delete their own todos
+
+    match /todos/{todoId} {
+      allow read: if request.auth.token.email_verified;
+      allow write: if request.auth.uid == request.resource.data.uid;
+      allow delete: if request.auth.uid == resource.data.uid;
```

Comment:
`write` consists of three different operations: create, update, and delete. You only need `request.resource.data.uid` for the create operation because, at the moment of creation, there is no `resource.data` to compare the request to. So a cleaner way to write these rules would be:
```suggestion
      allow create: if request.auth.uid == request.resource.data.uid;
      allow update, delete: if request.auth.uid == resource.data.uid;
```

Reply:
There is also a condition in the task: the users need to be authenticated to perform operations on their to-dos. To check this, we can use `request.auth != null`